File: VB/TreeList_SampleVB/Controllers/HomeController.vb
Imports System.Web.Mvc

Namespace Controllers
    Public Class HomeController
        Inherits Controller

        Public Function Index() As ActionResult
            Return View()
        End Function

        Public Function TreeListPartial(ByVal Expand? As Boolean) As ActionResult
            ViewData("Expand") = Expand
            Return PartialView(DepartmentsProvider.GetDepartments())
        End Function
    End Class
End Namespace

File: CS/TreeList_Sample/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using TreeList_Sample.Models;

namespace TreeList_Sample.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }
        public ActionResult TreeListPartial(bool? Expand) {
            ViewData["Expand"] = Expand;
            return PartialView(DepartmentsProvider.GetDepartments());
        }
    }
}